### PR TITLE
Add default dark mode

### PR DESCRIPTION
**Pull request by dbanksdesign**
#### Description of changes

Adding a new export `defaultDarkMode` to ui and ui-react packages that is of type `ColorModeOverride`. This allows customers to have dark mode support:

```jsx
  const theme: Theme = {
    name: 'my-theme',
    overrides: [defaultDarkMode],
  };
```

I have chosen to NOT include this dark mode override in the default theme so that it is opt-in. If we include the override in the default theme then customers would need to define an override to explicitly override the default override. It feels more backwards compatible and less scary to include it this way.

Update the dark mode theming docs and made the angular/vue dark mode docs more relevant. Ideally in the future we can export a dark mode CSS file that angular and vue could also export so their customers can have easy dark mode as well. 

We should probably wait to merge this after the docs changes @0618  is working on. 

#### Description of how you validated changes

Ran the docs site and the examples work as expected

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by 0618** [comment]
Thanks for waiting for https://github.com/aws-amplify/amplify-ui/pull/1871 ! ❤️

File: docs/src/pages/theming/dark-mode/examples/CustomDarkMode.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    <AmplifyProvider theme={theme} colorMode={colorMode}>
+      <Card>
+        <ToggleButtonGroup
+          value={colorMode}
+          isExclusive
+          onChange={(value: ColorMode) => setColorMode(value)}
+        >
+          <ToggleButton value="light">Light</ToggleButton>
+          <ToggleButton value="dark">Dark</ToggleButton>
+          <ToggleButton value="system">System</ToggleButton>
```
Comment (zchenwei) [inline docs/src/pages/theming/dark-mode/examples/CustomDarkMode.tsx:54]: Do you think we can remove this `system` toggle? We have a specific example for it.

File: docs/src/pages/theming/dark-mode/examples/DefaultDarkMode.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    <AmplifyProvider theme={theme} colorMode={colorMode}>
+      <Card>
+        <ToggleButtonGroup
+          value={colorMode}
+          isExclusive
+          onChange={(value: ColorMode) => setColorMode(value)}
+        >
+          <ToggleButton value="light">Light</ToggleButton>
+          <ToggleButton value="dark">Dark</ToggleButton>
+          <ToggleButton value="system">System</ToggleButton>
```
Comment (zchenwei) [inline docs/src/pages/theming/dark-mode/examples/DefaultDarkMode.tsx:30]: Same here

File: docs/src/pages/theming/dark-mode/examples/CustomDarkMode.tsx (hunk 1 above)
Comment (zchenwei) [reply docs/src/pages/theming/dark-mode/examples/CustomDarkMode.tsx:54]: NVM, I saw how you structure the docs

**Review by zchenwei — APPROVED** [review]
LGTM! :shipit:

File: packages/ui/src/theme/defaultDarkMode.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        // 60 doesn't change
+        80: { value: tokens.colors.neutral[40].value },
+        90: { value: tokens.colors.neutral[20].value },
+        100: { value: tokens.colors.neutral[10].value },
+      },
+      black: { value: '#fff' },
+      white: { value: '#000' },
+
+      border: {
+        primary: { value: '{colors.neutral.40.value}' },
```
Comment (zchenwei) [inline packages/ui/src/theme/defaultDarkMode.ts:97]: Nice! neutral-40 will meet the contrast ratio requirement in dark mode. I have a task that is related to some changes here. Would like to get this in ASAP so I can audit and work base off it.

File: docs/src/pages/theming/dark-mode/react.mdx [hunk 4]
```diff
@@ -1,3 +1,93 @@
...
 
+import {
+  DefaultDarkMode,
+  SystemDarkModeExample,
+  CustomDarkModeExample,
+} from './examples';
+
+## Overview
+
+Amplify UI supports color modes/schemes, like Dark Mode, through theme overrides. Amplify UI Theme Overrides let you define different theme styles in different contexts, such as color mode. You can also use plain CSS
```
Comment (reesscot) [inline docs/src/pages/theming/dark-mode/react.mdx:12]:
```suggestion
Amplify UI supports color modes/schemes, like Dark Mode, through theme overrides. Amplify UI Theme Overrides let you define different theme styles in different contexts, such as color mode. You can also use plain CSS.
```

File: docs/src/pages/theming/dark-mode/react.mdx [hunk 5]
```diff
@@ -1,3 +1,93 @@
...
+  <ExampleCode>
+    ```jsx file=./examples/DefaultDarkMode.tsx
+    ```
+    
+  </ExampleCode>
+</Example>
+
+### System Dark Mode
+
+If you don't want to provide a color mode control on your application, but still want to honor the user's operating system preference for color mode, you can set the `colorMode` on the AmplifyProvider to `system` and either use the default dark mode override or provide your own.
```
Comment (reesscot) [inline docs/src/pages/theming/dark-mode/react.mdx:53]: grammar nit: split into two sentences
```suggestion
If you don't want to provide a color mode control on your application, but still want to honor the user's operating system preference for color mode, you can set the `colorMode` on the AmplifyProvider to `system`. Then use either use the default dark mode override styling or provide your own.
```

File: docs/src/pages/theming/dark-mode/react.mdx [hunk 6]
```diff
@@ -1,3 +1,93 @@
...
+
+_Note: to see dark mode applied, change your OS preferences_
+
+<Example>
+  <SystemDarkModeExample />
+
+  <ExampleCode>
+    ```jsx file=./examples/SystemDarkMode.tsx
```
Comment (reesscot) [inline docs/src/pages/theming/dark-mode/react.mdx:61]:
```suggestion
    ```tsx file=./examples/SystemDarkMode.tsx
```

File: docs/src/pages/theming/dark-mode/react.mdx [hunk 7]
```diff
@@ -1,3 +1,93 @@
...
+    
+</ExampleCode>
+</Example>
+
+## CSS
+
+You can also write CSS variables directly to support color modes. If you are using the `colorMode` prop in the `AmplifyProvider` you will write CSS like this to support dark mode:
+
+```css
+@media (prefers-color-scheme: dark) {
```
Comment (reesscot) [inline docs/src/pages/theming/dark-mode/react.mdx:84]: Maybe add a helpful comment indicating that this detects the system setting?

File: packages/ui/src/theme/defaultDarkMode.ts [hunk 8]
```diff
@@ -0,0 +1,115 @@
+import { ColorModeOverride } from './types';
+import { tokens } from './tokens';
+
+/**
+ * A basic dark mode that just flips the base color
+ * palette.
+ */
+export const defaultDarkMode: ColorModeOverride = {
```
Comment (reesscot) [inline packages/ui/src/theme/defaultDarkMode.ts:8]: Since it's always passed to overrides, should it be called `defaultDarkModeOverride`?

**Review by reesscot — APPROVED** [review]
LGTM, just a few non-blocking suggestions

File: .changeset/gold-trains-help.md [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+"@aws-amplify/ui-react": minor
+"@aws-amplify/ui": minor
+---
+
+Add default dark mode theme override for React
+
+```jsx
+  const theme: Theme = {
+    name: 'my-theme',
+    overrides: [defaultDarkMode],
```
Comment (reesscot) [inline .changeset/gold-trains-help.md:11]:
```suggestion
    overrides: [defaultDarkModeOverride],
```

**Pull request merged**